keel-api-jackson/src/main/kotlin/com/netflix/spinnaker/keel/json/KeelApiModule.kt

```diff
@@ -55,13 +50,6 @@ object KeelApiModule : SimpleModule("Keel API") {
       setMixInAnnotations<Monikered, MonikeredMixin>()
       setMixInAnnotations<SubnetAwareRegionSpec, SubnetAwareRegionSpecMixin>()
       setMixInAnnotations<ConstraintState, ConstraintStateMixin>()
-
-      registerSubtypes(
```

Comment:
This is the only functional change in this PR. I did it to avoid a circular dependency between `keel-core` and `keel-artifact` via this module.

Reply:
Functional change as in, when this pr is merged then this functionality is removed?

Reply:
Yes, I just realized I can't merge this as is because it would break serialization of these sub-types until they're registered some other way. I'll think of something else.

Reply:
Based on this comment, I'm going to change approval status of my review.

Reply:
I've reverted that part of the PR. Please re-review.